src/Init/System/IO.lean

```diff
@@ -238,7 +238,12 @@ protected def TaskState.toString : TaskState → String
 
 instance : ToString TaskState := ⟨TaskState.toString⟩
 
-/-- Returns current state of the `Task` in the Lean runtime's task manager. -/
+/--
+Returns current state of the `Task` in the Lean runtime's task manager.
+
+Note that for tasks derived from `Promise`s, `waiting` and `running` should be considered
+equivalent.
```

Comment:
@tydeu I regret introduction of this API, it makes it impossible to hide implementation details based on `Task.map/bind`. This should have been some IO.Ref in Lake itself. But I think we'll survive.